src/main.py

```diff
@@ -156,12 +156,12 @@ def build_images(args):
 
     # We only upload to ECR once all images are successfully built locally.
     if args.target_ecr_repo is not None:
-        _push_images_upstream(image_versions, args.region)
+        _push_images_upstream(image_versions, args.region, args.is_public_ecr_repo)
 
 
-def _push_images_upstream(image_versions_to_push: list[dict[str, str]], region: str):
+def _push_images_upstream(image_versions_to_push: list[dict[str, str]], region: str, is_public_ecr_repo: bool = False):
     print(f'Will now push the images to ECR: {image_versions_to_push}')
-    username, password = _get_ecr_credentials(region)
+    username, password = _get_ecr_credentials(region, is_public_ecr_repo)
```

Comment:
ECR Public Gallery is global, so not sure if we need to pass `region` while instantiating the Boto3 client. Just double-check.

Reply:
Another thing to note here is that currently we get ECR credentials once and then use those to push all the images, so the assumption is that one credentials will work for all images/repositories.

Now, though, user could pass both private and public repositories at the same time, so the above assumption won't always hold true.

I think we'll have to move ECR credentials fetching inside the for-loop below.

Reply:
> I think we'll have to move ECR credentials fetching inside the for-loop below.
Done. Moved fetching ECR credentials inside the for loop.

> ECR Public Gallery is global, so not sure if we need to pass region while instantiating the Boto3 client. Just double-check.

It is always better to pass region . Else you might run into NoRegionError from botocore https://github.com/boto/botocore/blob/4c9cd012a1937fde04cd175218b1b1b7dafd78ef/botocore/regions.py#L277

In case if region is not provided, boto 3 tries to fetch the default region from aws config/environment variables.